metrics/metrics/src/test/java/io/helidon/metrics/TestServer.java

```diff
@@ -204,25 +212,16 @@ void checkMetricsForExecutorService() throws Exception {
 
         assertThat("Slow greet access response status", slowGreetResponse.status().code(), is(200));
 
-        MatcherWithRetry.retry(() -> {
-
-            WebClientResponse secondMetricsResponse = metricsRequestBuilder
-                    .submit()
-                    .await(CLIENT_TIMEOUT);
-
-            assertThat("Second access to metrics", secondMetricsResponse.status().code(), is(200));
-
-            JsonObject secondMetrics = secondMetricsResponse.content().as(JsonObject.class).await(CLIENT_TIMEOUT);
-
-            assertThat("JSON metrics results after accessing slow endpoint",
-                       secondMetrics,
-                       hasKey(jsonKeyForCompleteTaskCountInThreadPool));
-
-            int secondCompletedTaskCount = secondMetrics.getInt(jsonKeyForCompleteTaskCountInThreadPool);
-
-            assertThat("Completed task count after accessing slow endpoint", secondCompletedTaskCount, is(1));
-            return true;
-        });
+        retry.invoke(() -> metricsRequestBuilder
+                        .submit()
+                        .peek(res -> assertThat("Second access to metrics", res.status().code(), is(200)))
+                        .flatMapSingle(res -> res.content().as(JsonObject.class))
+                        .peek(json -> assertThat("JSON metrics results after accessing slow endpoint",
+                                json, hasKey(jsonKeyForCompleteTaskCountInThreadPool)))
+                        .map(json -> json.getInt(jsonKeyForCompleteTaskCountInThreadPool))
+                        .forSingle(secCompletedTaskCnt ->
+                                assertThat("Completed task count after accessing slow endpoint", secCompletedTaskCnt, is(1))))
+                .await(CLIENT_TIMEOUT);
```

Comment:
It is appealing to reuse the existing FT impl vs. a retry loop.

The JavaDoc for `peek` does say it's intended mainly for debugging and warns that the stream implementation might optimize away calls to `peek` for some elements of the stream. That probably cannot happen here (there are no `findFirst` or other optimizable steps in handling this stream).

Reply:
This is our reactive peek, we don't omit side-effects like Java streams does